pkg/soc/soc.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package swarm contains most basic and general Swarm concepts.
+package soc
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+	"github.com/ethersphere/bee/pkg/swarm"
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+)
+
+const (
+	IdSize        = 32
+	SignatureSize = 65
+	AddressSize   = crypto.AddressSize
+	minUpdateSize = IdSize + SignatureSize + swarm.SpanSize
+)
+
+// Id is a soc identifier
+type Id []byte
+
+// Owner is a wrapper that enforces valid length address of soc owner.
+type Owner struct {
+	address []byte
+}
+
+// NewOwner creates a new Owner.
+func NewOwner(address []byte) (*Owner, error) {
+	if len(address) != AddressSize {
+		return nil, fmt.Errorf("invalid address %x", address)
+	}
+	return &Owner{
+		address: address,
+	}, nil
+}
+
+// Update wraps a single soc update.
```

Comment:
how the notion of update made its way here.
I dont think there. is update on the level of soc